### PR TITLE
enhancement: making tooltips message be selected based on order status slug. changes to get_order_status_label filter

### PR DESCRIPTION
**Pull request by mino129**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

In a previous pull request https://github.com/woocommerce/woocommerce/pull/47861 a rework was introduced to how the tooltips in the WooCommerce Admin Order Listing works. 

As you can see in the new method `get_order_status_label` the message for the tooltip is retrieved from the `$status_names` array via checking if the label of the status (that is retrieved with the help of `wc_get_order_status_name`) is the same as the label in the arrays key. This has one big issue which is when a WordPress installation is not in English (and is in a language that has the WooCommerce strings translated), this will not work since the order status label's will be translated and will fail the retrieval from the array since the keys are different. 

In this PR I've changed the structure of the key-value pair of the `$status_names` array, making it be retrieved with the WooCommerce slugs for each order status, making translations a non-issue. I did not understand why this approach was not followed in the original PR, and if possible i would like to understand why! :)

Besides, in this pull request, i also added another change passing down the `WC_Order` object to the new filter `woocommerce_get_order_status_labels` so it allows for the tooltips message to be more customizable. 

For example, the old tooltips messages (tooltips before the original PR) we're absolutely "throw out" and one of the WooCommerce shops I run found the old tooltips useful. 

So in order for me to be able to customize the messages to make them the same as before, passing down the `WC_Order` object into the filter allows the messages to be customizable as one sees fits with the current order's info, for example making them the same as before.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check the tooltips on the WooCommerce Order Admin listing while the WordPress language is set to english and find that it works.
2. Change the WordPress language to anything else that's not English.
3. Go on the admin order listing page and you can find that the tooltips are no longer working due to the "bug" fixed in this PR.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Tooltips message is now selected based on the order status instead of the label of the order status, which would break if the current WordPress language was not English. Also passes the order object to the woocommerce_get_order_status_labels filter to allow for more personalized tooltips.

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php [hunk 1]
```diff
@@ -1064,14 +1064,14 @@ public function render_order_status_column( WC_Order $order ): void {
...
-			'On hold'         => __( 'The order is awaiting payment confirmation. Stock is reduced, but you need to confirm payment.', 'woocommerce' ),
-			'Processing'      => __( 'Payment has been received (paid), and the stock has been reduced. The order is awaiting fulfillment.', 'woocommerce' ),
-			'Completed'       => __( 'Order fulfilled and complete.', 'woocommerce' ),
-			'Failed'          => __( 'The customer’s payment failed or was declined, and no payment has been successfully made.', 'woocommerce' ),
-			'Draft'           => __( 'Draft orders are created when customers start the checkout process while the block version of the checkout is in place.', 'woocommerce' ),
-			'Canceled'        => __( 'The order was canceled by an admin or the customer.', 'woocommerce' ),
-			'Refunded'        => __( 'Orders are automatically put in the Refunded status when an admin or shop manager has fully refunded the order’s value after payment.', 'woocommerce' ),
+			'pending-payment' => __( 'The order has been received, but no payment has been made. Pending payment orders are generally awaiting customer action.', 'woocommerce' ),
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php]:
```suggestion
			'pending' => __( 'The order has been received, but no payment has been made. Pending payment orders are generally awaiting customer action.', 'woocommerce' ),
```

`pending` is the correct key name for Pending Payment status.

File: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php [hunk 2]
```diff
@@ -1064,14 +1064,14 @@ public function render_order_status_column( WC_Order $order ): void {
...
+			'on-hold'         => __( 'The order is awaiting payment confirmation. Stock is reduced, but you need to confirm payment.', 'woocommerce' ),
+			'processing'      => __( 'Payment has been received (paid), and the stock has been reduced. The order is awaiting fulfillment.', 'woocommerce' ),
+			'completed'       => __( 'Order fulfilled and complete.', 'woocommerce' ),
+			'failed'          => __( 'The customer’s payment failed or was declined, and no payment has been successfully made.', 'woocommerce' ),
+			'draft'           => __( 'Draft orders are created when customers start the checkout process while the block version of the checkout is in place.', 'woocommerce' ),
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php]:
```suggestion
			'checkout-draft'           => __( 'Draft orders are created when customers start the checkout process while the block version of the checkout is in place.', 'woocommerce' ),
```

`checkout-draft` is the correct key name for Draft status.

**Review by adrianduffell — CHANGES_REQUESTED** [review]
Thanks for catching this @mino129 and contributing a fix . It tested well for me, with the minor exception of a couple of changes needed for the "Pending payment" and "Draft" keys.

We can include this in the next WooCommerce release however we need to merge the PR today before the code freeze.

> In this PR I've changed the structure of the key-value pair of the $status_names array, making it be retrieved with the WooCommerce slugs for each order status, making translations a non-issue. I did not understand why this approach was not followed in the original PR, and if possible i would like to understand why! :)

We've simply missed this in our testing. I'm sorry for the inconvenience 😅

> So in order for me to be able to customize the messages to make them the same as before, passing down the WC_Order object into the filter allows the messages to be customizable as one sees fits with the current order's info, for example making them the same as before.

I think other users might also like this customization to restore the old tooltips. If you're interested, it would be great to share it as a code snippet or even a plugin on wordpress.org.

**Comment by mino129** [comment]
Hello @adrianduffell !

Thank you for your feedback on the PR, I've gone ahead and implemented the changes suggested and push another commit onto this PR with the changes! :)

Please let me know if further changes are needed.

> I think other users might also like this customization to restore the old tooltips. If you're interested, it would be great to share it as a code snippet or even a plugin on wordpress.org.

I'll gladly create a gist with this customization once this change is live in the core plugin! :)

Thank you for your help and if anything else is needed please do let me know!

**Comment by adrianduffell** [comment]
Thanks for pushing the latest changes @mino129 , the pending and draft statuses work well for me now!

It looks like one more change and then we can merge, could you please remove one of the changelog files? 🙏 

> I'll gladly create a gist with this customization once this change is live in the core plugin! :)

This would be perfect, thank you! I'll be happy to test this out if you ping me.

**Comment by mino129** [comment]
Hey @adrianduffell , sorry totally missed that!

I've deleted the duplicate file and pushed a commit with the change!

Please let me know if you need anything else! :)

**Review by adrianduffell — APPROVED** [review]

**Comment by chihsuan** [comment]
Hey @mino129 Lint job is failing on this PR. Can you take a look at it?  Let us know if you need help with it. I can push a commit to fix it if you want.


```
. lint:changes:branch:php: Linting the following files with CodeSniffer:
. lint:changes:branch:php:     src/Internal/Admin/Orders/ListTable.php
. lint:changes:branch:php: FILE: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php: FOUND 0 ERRORS AND 8 WARNINGS AFFECTING 8 LINES
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php:  1067 | WARNING | Array double arrow not aligned correctly; expected 8 space(s) between "'pending'" and double arrow, but found 1. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1068 | WARNING | Array double arrow not aligned correctly; expected 8 space(s) between "'on-hold'" and double arrow, but found 9. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1069 | WARNING | Array double arrow not aligned correctly; expected 5 space(s) between "'processing'" and double arrow, but found 6. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1070 | WARNING | Array double arrow not aligned correctly; expected 6 space(s) between "'completed'" and double arrow, but found 7. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1071 | WARNING | Array double arrow not aligned correctly; expected 9 space(s) between "'failed'" and double arrow, but found 10. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1072 | WARNING | Array double arrow not aligned correctly; expected 1 space(s) between "'checkout-draft'" and double arrow, but found 11. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1073 | WARNING | Array double arrow not aligned correctly; expected 6 space(s) between "'cancelled'" and double arrow, but found 7. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php:  1074 | WARNING | Array double arrow not aligned correctly; expected 7 space(s) between "'refunded'" and double arrow, but found 8. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php: Script phpcs-changed handling the __exec_command event returned with error code 1
. lint:changes:branch:php: Unfortunately, CodeSniffer reported some violations which need to be addressed (see above).
```

**Comment by mino129** [comment]
hey @chihsuan , thanks for the warning i forgot to run the lint check before the last commit.

I've gone ahead and fixed the linting issues reported.

Please let me know if anything else is needed.

**Comment by adrianduffell** [comment]
Thanks @mino129, this is all set for the 9.2 release on Aug 14!

**Comment by ilyasfoo** [comment]
> I'll gladly create a gist with this customization once this change is live in the core plugin! :)

Hi, @mino129! 👋 

Appreciate your work here! I'm wondering if you're still interested to create a gist? We could use a [9.2 RC.1 build](https://github.com/woocommerce/woocommerce/releases/tag/9.2.0-rc.1) to test it prior the official release.

**Comment by mino129** [comment]
Hi @ilyasfoo!

Thanks for your reminder on the gist. I already have something quite close to this working on a production WooCommerce installation, so I've gone ahead and made some changes to make sure it works with the changes in this PR.

https://gist.github.com/mino129/0e5d7ee44ddbba4fb4a7162a4038549d

Please let me know if anything else is needed! :)

**Comment by ilyasfoo** [comment]
Thank you, @mino129! I was able to use your code as a base and asked AI to improve, lint, and wrap it as a plugin.

However, translations no longer work since the original strings are removed from WooCommerce plugin.

I've forked your gist, feel free to suggest edits or improve it.

https://gist.github.com/ilyasfoo/2d614e01b2100f2b476e235bd4cdba06

**Pull request merged**